### PR TITLE
Array sort

### PR DESCRIPTION
**Pull request by szuend**
PR for initial review.

@mathiasbynens

### CODE REVIEW

File: spec.html [hunk 1]
```diff
@@ -32544,67 +32544,60 @@ <h1>Array.prototype.some ( _callbackfn_ [ , _thisArg_ ] )</h1>
...
+          1. Let _j_ be 0.
+          1. Repeat, while _j_ &lt; _itemCount_
+            1. Perform ? Set(_obj_, ! ToString(_j_), items[_j_], *true*).
+            1. Set _j_ to _j_ + 1.
+          1. Repeat, while _j_ &lt; _itemCount_ + _undefinedCount_
+            1. Perform ? Set(_obj_, ! ToString(_j_), *undefined*, *true*).
+            1. Set _j_ to _j_ + 1.
+          1. Repeat, while _j_ &lt; len
+            1. Let _Pj_ be ! ToString(_j_).
+            1. Let _jPresent_ be ? HasProperty(_obj_, _Pj_).
```
Comment (mathiasbynens) [inline spec.html]: As discussed offline, this can be HasOwnProperty

File: spec.html [hunk 2]
```diff
@@ -32544,67 +32544,60 @@ <h1>Array.prototype.some ( _callbackfn_ [ , _thisArg_ ] )</h1>
...
-          </li>
-          <li>
-            If _comparefn_ is *undefined* and the application of ToString to any value passed as an argument to SortCompare modifies _obj_ or any object on _obj_'s prototype chain.
-          </li>
-          <li>
-            If _comparefn_ is *undefined* and all applications of ToString, to any specific value passed as an argument to SortCompare, do not produce the same result.
+            If _comparefn_ is *undefined*:
+            <ul>
+              <li>
+                SortCompare does not act as a consitent comparison function.
```
Comment (mathiasbynens) [inline spec.html]: consistent

File: spec.html [hunk 3]
```diff
@@ -32544,67 +32544,60 @@ <h1>Array.prototype.some ( _callbackfn_ [ , _thisArg_ ] )</h1>
...
+              </li>
+              <li>
+                All applications of ToString, to any specific value passed as an argument to SortCompare, do not produce the same result.
+              </li>
```
Comment (mathiasbynens) [inline spec.html]: For clarity and simplicity, I'd avoid the nested list here and instead flatten it:

```html
<li>If _comparefn_ is not *undefined* and is not a consistent comparison function for the elements of _items_ (see below).</li>
<li>If _comparefn_ is *undefined* and ...</li>
<li>If _comparefn_ is *undefined* and ...</li>
```

File: spec.html (hunk 1 above)
Comment (szuend) [reply spec.html]: Done

File: spec.html (hunk 2 above)
Comment (szuend) [reply spec.html]: Done

File: spec.html (hunk 3 above)
Comment (szuend) [reply spec.html]: Done

File: spec.html [hunk 4]
```diff
@@ -32544,67 +32544,55 @@ <h1>Array.prototype.some ( _callbackfn_ [ , _thisArg_ ] )</h1>
...
+          1. Let _undefinedCount_ be 0.
+          1. Repeat, while _k_ &lt; _len_
+            1. Let _Pk_ be ! ToString(_k_).
+            1. Let _kPresent_ be ? HasProperty(_obj_, _Pk_).
+            1. If _kPresent_ is *true*, then
+              1. Let _kValue_ be ? Get(_obj_, _Pk_).
+              1. If _kValue_ is *undefined*, then
+                1. Set _undefinedCount_ to _undefinedCount_ + 1.
+              1. Else,
+                1. Set _items_[_itemCount_] to _kValue_.
```
Comment (mathiasbynens) [inline spec.html]: _kValue_ is not defined in this case

**Pull request closed without merging**